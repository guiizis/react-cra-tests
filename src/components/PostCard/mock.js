export const postCardPropsMock = {
  title: 'test 1',
  body: 'test 2',
  id: '1',
  cover: 'img/img.png'
};
