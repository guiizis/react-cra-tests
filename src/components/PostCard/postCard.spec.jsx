import { render, screen } from '@testing-library/react';
import { PostCard } from '.';
import { postCardPropsMock as props } from './mock';

describe('<PostCard/>', () => {
  it('should render postCard correctly', () => {
    render(<PostCard {...props} />);

    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute('src', props.cover);
    expect(screen.getByRole('img', { name: props.title })).toHaveAttribute('src', props.cover);
    expect(screen.getByText('test 2')).toBeInTheDocument();
  });

  it('should match snapshot', () => {
    const { container } = render(<PostCard {...props} />);

    expect(container.firstChild).toMatchSnapshot();
  });
});
