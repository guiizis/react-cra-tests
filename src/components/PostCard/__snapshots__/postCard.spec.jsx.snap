// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard/> should match snapshot 1`] = `
<div
  class="post"
>
  <img
    alt="test 1"
    src="img/img.png"
  />
  <div
    class="post-content"
  >
    <h2>
      test 1
    </h2>
    <h5>
      test 2
    </h5>
  </div>
</div>
`;
